src/lib/schematics/update/rules/checkImportMiscRule.ts

```diff
@@ -12,26 +12,43 @@ import * as ts from 'typescript';
 import {isMaterialImportDeclaration} from '../material/typescript-specifiers';
 
 /**
- * Rule that walks through every identifier that is part of Angular Material and replaces the
- * outdated name with the new one.
+ * Rule that detects import declarations that refer to outdated identifiers from Angular Material
+ * or the CDK which cannot be updated automatically.
  */
 export class Rule extends Rules.TypedRule {
   applyWithProgram(sourceFile: ts.SourceFile, program: ts.Program): RuleFailure[] {
-    return this.applyWithWalker(new CheckImportMiscWalker(sourceFile, this.getOptions(), program));
+    return this.applyWithWalker(new Walker(sourceFile, this.getOptions(), program));
   }
 }
 
-export class CheckImportMiscWalker extends ProgramAwareRuleWalker {
-  visitImportDeclaration(declaration: ts.ImportDeclaration) {
-    if (isMaterialImportDeclaration(declaration)) {
-      declaration.importClause.namedBindings.forEachChild(n => {
-        let importName = n.getFirstToken() && n.getFirstToken().getText();
-        if (importName === 'SHOW_ANIMATION' || importName === 'HIDE_ANIMATION') {
-          this.addFailureAtNode(
-              n,
-              `Found deprecated symbol "${red(importName)}" which has been removed`);
-        }
-      });
+export class Walker extends ProgramAwareRuleWalker {
+
+  visitImportDeclaration(node: ts.ImportDeclaration) {
+    if (!isMaterialImportDeclaration(node) ||
+        !node.importClause ||
+        !node.importClause.namedBindings) {
+      return;
+    }
+
+    const namedBindings = node.importClause.namedBindings;
```

Comment:
Note: Storing the `namedBindings` here because future "import misc" cases will also _always_ depend on `namedBindings`.